app/ghcup/Main.hs

```diff
@@ -68,17 +69,18 @@ import qualified GHCup.Types                   as Types
 toSettings :: Options -> IO (Settings, KeyBindings, UserSettings)
 toSettings options = do
   noColor <- isJust <$> lookupEnv "NO_COLOR"
+  pagerCmd <- getPager
```

Comment:
We're unconditionally executing this, which is a bit unfortunate... I think lazy IO could be fine here.